pkg/client/unversioned/conditions.go

```diff
@@ -17,11 +17,46 @@ limitations under the License.
 package unversioned
 
 import (
+	"time"
+
 	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/apis/extensions"
 	"k8s.io/kubernetes/pkg/util/wait"
 )
 
+// DefaultBackoff is the recommended client side backoff for a conflict.
+var DefaultBackoff = wait.Backoff{
+	Steps:    3,
+	Duration: 10 * time.Millisecond,
+	Factor:   10.0,
+	Jitter:   0.1,
+}
+
+// RetryConflict executes the provided function repeatedly, retrying if the server returns a conflicting
+// write. Callers should preserve previous executions if they wish to retry changes. It performs an
+// exponential backoff.
+// TODO: Make Backoff an interface?
+func RetryOnConflict(backoff wait.Backoff, fn func() error) error {
+	var lastConflictErr error
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
```

Comment:
I was torn on this (and the default backoff may not be the right numbers).  A common case we're wanting to use this in is where multiple clients are competing to update the same entry.  We have a few controllers that are performing idempotent actions on the same resource, but we also have actions driven by clients where the initial action triggers a flurry of updates.

For example, our deployment controller (in OpenShift) deals with the deployment trigger controller updating the same resources.  That's a case where straight up retry is appropriate.  The second case is creating a deployment from the CLI - the first action (create) results in a bunch of controllers acting on it to create status.  In that case, backoff helps reduce the conflict.

I can add a `DefaultRetry` that has Factor 1.0.  I'll also find one of the instances in the code and replace it.